shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/presentation/modules/surveyselection/SurveySelectionViewModel.kt

```diff
@@ -42,7 +52,41 @@ class SurveySelectionViewModel(
                 .combine(getAppVersion()) { userDate, version ->
                     Triple(userDate.first, userDate.second, version)
                 }
-                .collect { updateHeaderState(it.first, it.second, it.third) }
+                .combine(fetchSurvey(currentPage)) { userData, survey ->
+                    Pair(userData, survey)
+                }
+                .collect {
+                    updateHeaderState(it.first.first, it.first.second, it.first.third)
+                    updateSurveyState(it.second)
+                }
+        }
+    }
+
+    fun checkFetchMore(item: Int) {
```

Comment:
@blyscuit Is it `itemPosition` or `itemIndex`? 🤔

Reply:
Nice catch.